apis/v1/gatewayclass_types.go

```diff
@@ -180,9 +180,12 @@ const (
 	GatewayClassReasonAccepted GatewayClassConditionReason = "Accepted"
 
 	// This reason is used with the "Accepted" condition when the GatewayClass
-	// was not accepted because the parametersRef field refers to a nonexistent
-	// or unsupported resource or kind, or when the data within that resource is
-	// malformed.
+	// was not accepted because the parametersRef field refers to
+	// * a namespaced resource but the Namespace field is not set, or
+	// * a cluster-scoped resource but the Namespace field is set, or
+	// * a nonexistent object, or
+	// * an unsupported resource or kind, or
+	// * a legit and existent resource but the data within that resource is malformed.
```

Comment:
Tiny nit: This might be a bit simpler:
```suggestion
	// * an existing resource but the data within that resource is malformed.
```